internal/server/instance/drivers/driver_qemu.go

```diff
@@ -1653,7 +1653,16 @@ func (d *qemu) start(stateful bool, op *operationlock.InstanceOperation) error {
 
 	// Attempt to drop privileges (doesn't work when restoring state).
 	if !stateful && d.state.OS.UnprivUser != "" {
-		qemuArgs = append(qemuArgs, "-runas", d.state.OS.UnprivUser)
+		qemuVer91, _ := version.NewDottedVersion("9.1.0")
+		qemuVer, _ := d.version()
+		if qemuVer != nil {
+			// Since QEMU 9.1 the parameter `runas` has been marked as deprecated.
+			if qemuVer.Compare(qemuVer91) >= 0 {
+				qemuArgs = append(qemuArgs, "-run-with", fmt.Sprintf("user=%s", d.state.OS.UnprivUser))
+			} else {
+				qemuArgs = append(qemuArgs, "-runas", d.state.OS.UnprivUser)
+			}
+		}
```

Comment:
In this case, if the version doesn't parse we end up with nothing.
I think we should instead be doing:

```go
if qemuVer != nil && qemuVer.Compare(qemuVer91) >= 0 {
// use run-with
} else {
// use runas
}
```

Reply:
That also has the benefit of making things a bit more compact and easier to read

Reply:
Fixed on https://github.com/lxc/incus/pull/1871/commits/6eea371007bdea422d493b16e7ce0677aa709c36